mono/mini/interp/interp.h

```diff
@@ -28,48 +28,10 @@ struct _InterpMethodArguments {
 
 typedef struct _InterpMethodArguments InterpMethodArguments;
 
-typedef struct _MonoInterpStackIter MonoInterpStackIter;
-
-/* Needed for stack allocation */
-struct _MonoInterpStackIter {
-	gpointer dummy [8];
-};
-
-typedef gpointer MonoInterpFrameHandle;
-
-
-struct _MonoInterpCallbacks {
-	gpointer (*create_method_pointer) (MonoMethod *method, MonoError *error);
-	MonoObject* (*runtime_invoke) (MonoMethod *method, void *obj, void **params, MonoObject **exc, MonoError *error);
-	void (*init_delegate) (MonoDelegate *del);
-#ifndef DISABLE_REMOTING
-	gpointer (*get_remoting_invoke) (gpointer imethod, MonoError *error);
-#endif
-	gpointer (*create_trampoline) (MonoDomain *domain, MonoMethod *method, MonoError *error);
-	void (*walk_stack_with_ctx) (MonoInternalStackWalk func, MonoContext *ctx, MonoUnwindOptions options, void *user_data);
-	void (*set_resume_state) (MonoJitTlsData *jit_tls, MonoException *ex, MonoJitExceptionInfo *ei, MonoInterpFrameHandle interp_frame, gpointer handler_ip);
-	gboolean (*run_finally) (StackFrameInfo *frame, int clause_index, gpointer handler_ip);
-	gboolean (*run_filter) (StackFrameInfo *frame, MonoException *ex, int clause_index, gpointer handler_ip);
-	void (*frame_iter_init) (MonoInterpStackIter *iter, gpointer interp_exit_data);
-	gboolean (*frame_iter_next) (MonoInterpStackIter *iter, StackFrameInfo *frame);
-	MonoJitInfo* (*find_jit_info) (MonoDomain *domain, MonoMethod *method);
-	void (*set_breakpoint) (MonoJitInfo *jinfo, gpointer ip);
-	void (*clear_breakpoint) (MonoJitInfo *jinfo, gpointer ip);
-	MonoJitInfo* (*frame_get_jit_info) (MonoInterpFrameHandle frame);
-	gpointer (*frame_get_ip) (MonoInterpFrameHandle frame);
-	gpointer (*frame_get_arg) (MonoInterpFrameHandle frame, int pos);
-	gpointer (*frame_get_local) (MonoInterpFrameHandle frame, int pos);
-	gpointer (*frame_get_this) (MonoInterpFrameHandle frame);
-	void (*frame_arg_to_data) (MonoInterpFrameHandle frame, MonoMethodSignature *sig, int index, gpointer data);
-	void (*data_to_frame_arg) (MonoInterpFrameHandle frame, MonoMethodSignature *sig, int index, gpointer data);
-	gpointer (*frame_arg_to_storage) (MonoInterpFrameHandle frame, MonoMethodSignature *sig, int index);
-	MonoInterpFrameHandle (*frame_get_parent) (MonoInterpFrameHandle frame);
-	void (*start_single_stepping) (void);
-	void (*stop_single_stepping) (void);
-};
-
-void mono_interp_parse_options (const char *options);
-
-void mono_interp_init (void);
+/* must be called either
+ *  - by mini_init ()
+ *  - xor, before mini_init () is called (embedding scenario).
```

Comment:
Was this supposed to be 'or'?

Reply:
no 🙂  `mono_interp_init ()` is supposed to be called once, otherwise it will assert.

Reply:
"execution" makes more sense than "compilation", as a way to abstract interpreter, JIT, AOT -- they are executing one way or another, but they aren't necessarily compiling. Is VES/EE the larger system, like the entire runtime? Perhaps nobody has specified a finer grained break down? You know, to do that, would require stuff like talking about how the pieces are connected, you can't split up gc and jit/interp/aot w/o talking about barriers and suspend. If you smush everything together architecturally/"documentationally" then from the outside you can just say it is magic. And truly, from the "C#" point of view, that is perhaps the right level of specification. Implementers can chose to smush it all together and have one of everything, intertwined, or have plugability of JIT/interp/AOT or maybe other parts. What do the .NET Native people call things? Or the IL2CPP people?

Reply:
I happen to like "execution engine", but we don't have a specific name for something like this from the IL2CPP side.